sphinx_gallery/notebook.py

```diff
@@ -142,36 +145,46 @@ def add_code_cell(work_notebook, code):
     work_notebook["cells"].append(code_cell)
 
 
-def add_markdown_cell(work_notebook, text):
+def add_markdown_cell(work_notebook, markdown):
     """Add a markdown cell to the notebook
 
     Parameters
     ----------
-    code : str
-        Cell content
+    markdown : str
+        Markdown cell content.
     """
     markdown_cell = {
         "cell_type": "markdown",
         "metadata": {},
-        "source": [rst2md(text)]
+        "source": [markdown]
     }
     work_notebook["cells"].append(markdown_cell)
 
 
-def fill_notebook(work_notebook, script_blocks):
+def fill_notebook(work_notebook, script_blocks, gallery_conf):
     """Writes the Jupyter notebook cells
 
+    If available, uses pypandoc to convert rst to markdown.
+
     Parameters
     ----------
     script_blocks : list
         Each list element should be a tuple of (label, content, lineno).
     """
-
     for blabel, bcontent, lineno in script_blocks:
         if blabel == 'code':
             add_code_cell(work_notebook, bcontent)
         else:
-            add_markdown_cell(work_notebook, bcontent + '\n')
+            if gallery_conf["pypandoc"] == False:
+                markdown = rst2md(bcontent + '\n')
+            else:
+                import pypandoc
+                # pandoc automatically addds \n to the end
+                markdown = pypandoc.convert_text(
+                    bcontent, to='md', format='rst', **gallery_conf["pypandoc"]
+                )
+
+        add_markdown_cell(work_notebook, markdown)
```

Comment:
This still needs to be indented one level deeper (to match old behavior at least)

Reply:
whoops, thanks